api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentService.java

```diff
@@ -42,8 +45,12 @@ public Payment makeFirstPayment(PaymentCycle paymentCycle, String cardAccountId)
     }
 
     public Payment makePayment(PaymentCycle paymentCycle, String cardAccountId) {
-        Integer amountToPay = checkBalanceAndCalculatePaymentAmount(paymentCycle);
-        // TODO: HTBHF-1267: If payment amount < minimum payment amount then don't make a payment (might need a new payment status)
+        Integer amountToPay = checkBalanceAndCalculatePaymentAmount(paymentCycle, cardAccountId);
+        if (amountToPay == 0) {
+            //TODO HTBHF-1418: Create an event for no payment made due to balance too high.
+            log.info("No payment will be made as the existing balance on the card is too high");
+            return null;
+        }
```

Comment:
We also need to update the PaymentCycle with the amount paid (including if the amount is zero)

Reply:
Hmmm. Just noticed that we don't have a payment amount on PaymentCycle. Leave it out of this PR then. We'll want to have a think about how we distinguish between incomplete payments and those unpaid due to card balance. I guess if the PaymentCycle has a card balance amount then it can be considered complete, and if there'sno payment it is because the balance was too high.

Reply:
OK, I will add a subtask so we can discuss the best option next week